src/online/unigrid.jl

```diff
@@ -43,3 +43,40 @@ function updateZ(rng::AbstractRNG, Z::AbstractVector, alg::UniGrid, X::AbstractV
     Zn = deepcopy(Z)
     return updateZ!(rng, Zn, alg, X)
 end
+
+export UniformGrid
+struct UniformGrid{N, T} <: AbstractVector{T} 
+    proditer::Iterators.ProductIterator{NTuple{N, LinRange{T, Int64}}}
+
+    function UniformGrid(proditer::Iterators.ProductIterator{NTuple{N, LinRange{T, Int64}}}) where {N, T}
+        new{N,T}(proditer)
+    end
+end
+
+import Base: getindex, broadcastable, eachindex, length, size, enumerate
+Base.getindex(ug::UniformGrid, i) = collect(first(Iterators.drop(ug.proditer, i-1)))
+
+Base.broadcastable(ug::UniformGrid) = Base.broadcastable(ug.proditer)[:]
+
+Base.eachindex(ug::UniformGrid) = Base.OneTo(length(ug))
+
+Base.length(ug::UniformGrid) = prod(length.(ug.proditer.iterators))
+# alternative: (typeof(t).parameters[1], prod(length.(ug.proditer.iterators)))
+Base.size(ug::UniformGrid) = length.(ug.proditer.iterators)
```

Comment:
I think `size` needs to return a `Tuple` ?

Reply:
It does. `Iterators.Product{...}.iterators` returns a `Tuple` of iterators, and broadcasting `length` keeps it a tuple.

Reply:
Ah yeah, but that's not what we want. We return an `AbstractVector` so it should be the same as `length`
```suggestion
Base.size(ug::UniformGrid) = (prod(length, ug.proditer.iterators),)
```

Reply:
No problem, I wasn't quite sure what would be correct here, so I just put in something.

Reply:
We need to consider this as an `AbstractVector{<:AbstractVector}` so the size of it should be the total number of points.